typescript/test/services/deposits.test.ts

```diff
@@ -947,7 +948,124 @@ describe("Deposits", () => {
   })
 
   describe("DepositsService", () => {
-    // TODO: Implement unit tests.
+    describe("initiateDeposit", () => {
+      const depositor = EthereumAddress.from(
+        "934b98637ca318a4d6e7ca6ffd1690b8e77df637"
+      )
+      const bitcoinClient = new MockBitcoinClient()
+      const tbtcContracts = new MockTBTCContracts()
+      let depositService: DepositsService
+
+      beforeEach(async () => {
+        depositService = new DepositsService(tbtcContracts, bitcoinClient)
+      })
+
+      context("when default depositor is not set", () => {
+        it("should throw", async () => {
+          await expect(
+            depositService.initiateDeposit("mjc2zGWypwpNyDi4ZxGbBNnUA84bfgiwYc")
+          ).to.be.rejectedWith(
+            "Default depositor is not set; use setDefaultDepositor first"
+          )
+        })
+      })
+
+      context("when default depositor is set", () => {
+        beforeEach(async () => {
+          depositService.setDefaultDepositor(depositor)
+        })
+
+        context("when active wallet is not set", () => {
+          it("should throw", async () => {
+            await expect(
+              depositService.initiateDeposit(
+                "mjc2zGWypwpNyDi4ZxGbBNnUA84bfgiwYc"
+              )
+            ).to.be.rejectedWith("Could not get active wallet public key")
+          })
+        })
+
+        context("when active wallet is set", () => {
+          beforeEach(async () => {
+            tbtcContracts.bridge.setActiveWalletPublicKey(
+              Hex.from(
+                "03989d253b17a6a0f41838b84ff0d20e8898f9d7b1a98f2564da4cc29dcf8581d9"
+              )
+            )
+          })
+
+          context("when recovery address is incorrect", () => {
+            it("should throw", async () => {
+              await expect(
+                depositService.initiateDeposit(
+                  "2N5WZpig3vgpSdjSherS2Lv7GnPuxCvkQjT" // p2sh address
+                )
+              ).to.be.rejectedWith(
+                "Bitcoin recovery address must be P2PKH or P2WPKH"
+              )
+            })
+          })
+
+          context("when recovery address is correct", () => {
+            context("when recovery address is P2PKH", () => {
+              let deposit: Deposit
+
+              beforeEach(async () => {
+                deposit = await depositService.initiateDeposit(
+                  "mjc2zGWypwpNyDi4ZxGbBNnUA84bfgiwYc"
+                )
+              })
+
+              it("should initiate deposit correctly", async () => {
+                // Inspect the deposit object by looking at its receipt.
+                const receipt = deposit.getReceipt()
+                expect(receipt.depositor).to.be.equal(depositor)
```

Comment:
Can we make sure the blinding factor is set and has 8 bytes? Same for the case below.

Reply:
Done in 1fcc4c92bdd0afb093e1d35a9876bd9e796473ff.